### ISSUE TITLE
Potential Dispatch regression - QnA overly weighted against LUIS?

### ISSUE THREAD

**Issue by darrenj** [Microsoft/botbuilder-tools#1101]
## Tool
Name: Dispatch
Version:1.4.4-preview-190425-01
OS: Windows

## Describe the bug
I think we might have a regression with Dispatch evaluation with the latest build (1.4.4-preview-190425-01). We have changed our use of Dispatch so not ruling that out but suspect it might be in how Dispatch is working (or LUIS of course!). Instead of grouping all utterances into “l_Calendar” or l_Email we now split into l_BookMeeting, l_MeetingSummary to best align with actions.  As below:

![image](https://user-images.githubusercontent.com/33519520/56848698-da3ff400-68e3-11e9-94d6-12ba16097997.png)

My favourite testing utterance for calendar is “what meetings do I have”, for some very odd reason Dispatch model is suggesting q_faq and q_chitchat as the highest ranking despite them not appearing to have anything relating to that in the training data. Echoed by the fact that QnAMaker doesn’t return anything!

![image](https://user-images.githubusercontent.com/33519520/56848705-e926a680-68e3-11e9-8c4c-2ee22d4d970f.png)

When comparing to a previous Dispatch model with all calendar utterances grouped together you can see the difference from before to now. QnA gets a very low score?

![image](https://user-images.githubusercontent.com/33519520/56848711-ffccfd80-68e3-11e9-8809-e51f2b4a67f1.png)

[bug]

**Comment by tsuwandy** [comment]
@darrenj, added Hung-chih's comment below.  

Short summary:
------------------
For the first 2 utterances ("what meetings do i have", "what meetings") the models do not have confidence in their predictions, so please add them or some similar utterances to one of the intent training set.

For the third, the apostrophe is more prevalent in q_chitchat than in other intents. Since apostrophe is not really semantically linked to any intent, it would be better either to 1) add this feature (such as “what’s”) to all intents), or 2) remove it from all of them.

Let us know if you are able to make adjustments to the Dispatch per the recommendations above. Will keep this open till the Dispatch model is working as expected.

From: Hung-chih Yang
Sent: Saturday, April 27, 2019 9:32:20 PM

Hi Darren,

I used Dispatch to directly train and evaluate the LUIS app, djvakona2en_Dispatch.json, with useAllTrainingData flag turned on.

Here are my observations:
1)	For utterance “what meetings do I have”
a.	Score for ‘q_chitchat’ is 0.2969, the highest, while the score for the expected intent ‘calendarskill_summary’ is 0.1068.
b.	Though LUIS team recommends serving the top intent, ‘q_chitchat’, as the prediction, such suggestion is counter to common data science practice as both scores are very low, so the models do not have confidence in either predictions. Usually a data scientist would set up an acceptance threshold, say 0.5. The exact value is application dependent based on a data scientist’s judgement. If a prediction is lower than that threshold, the prediction should not be used.
c.	Between ‘q_chitchat’ and ‘calendarskill_summary’, the features that distract the prediction away from ‘calendarskill_summary’ towards ‘q_chitchat’ are (ordered by a distraction score):
i.	‘i’ with a distraction score: 0.2349
ii.	‘do’: 0.20735
iii.	‘have’: 0.0021
iv.	‘what’: -0.04048
v.	‘meetings’: -0.133086
d.	In short, even though ‘meetings is a positive feature/signal for ‘calendarskill_summary’, its effect is not strong enough to counter the effect of ‘I’, ‘do’, and ‘have’ toward ‘q_chitchat’.
2)	For “what meetings”
a.	The top predicted intent is ‘q_general’ with a score of 0.24935, while the score for the expected intent ‘calendarskill_summary’ is 0.12524
b.	Again, the scores for either is very small, thus the models have no confidence in these predictions.
3)	For “what’s on today”
a.	Score for ‘q_chitchat’ is 0.97092, the highest, while the score for the expected intent ‘calendarskill_summary’ is 0.000212.
b.	The features “what”, “’”, “on”, “today” are stronger on ‘q_chitchat’ 
c.	‘q_chitchat’ has an utterance “what's going on”, which is very close to “what’s on today”
d.	The “’” feature has the strongest distraction score (0.4539) toward ‘q_chitchat’.
e.	In short, the apostrophe “’” feature is very strong in q_chitchat, but not so much in ‘calendarskill_summary’. The follow-up is “on”.

In summary, for the first 2 utterances, the models do not have confidence in their predictions, so please add them or some similar utterances to one of the intent training set.
For the third, the apostrophe is more prevalent in q_chitchat than in other intents. Since apostrophe is not really semantically linked to any intent, it would be better either to 1) add this feature (such as “what’s”) to all intents), or 2) remove it from all of them.

Hung-chih

**Comment by tsuwandy** [comment]
Not Dispatch regression, closing.
